tests/sentry/models/test_notificationaction.py

```diff
@@ -0,0 +1,71 @@
+from collections import defaultdict
+from unittest.mock import MagicMock, patch
+
+import pytest
+from django.forms import ValidationError
+
+from sentry.models.notificationaction import NotificationAction
+from sentry.models.notificationaction import logger as NotificationActionLogger
+from sentry.testutils import TestCase
+
+
+class NotificationActionTest(TestCase):
+    def setUp(self):
+        self.organization = self.create_organization(name="night city")
+        self.projects = [
+            self.create_project(name="netrunner", organization=self.organization),
+            self.create_project(name="edgerunner", organization=self.organization),
+        ]
+        self.notif_action = self.create_notification_action(
+            organization=self.organization, projects=self.projects
+        )
+        self.illegal_trigger = (-1, "sandevistan")
+        self.illegal_service = (-1, "braindance")
+        # Reset registers for tests
+        NotificationAction._handlers = defaultdict(dict)
```

Comment:
This will unregister all handlers and not restore them, so will leak into other tests.. I'd recommend using `mock.patch.dict` to clear it and reset back to default after. You can do something similar to what I did here: https://github.com/getsentry/getsentry/pull/9780/files#diff-30af88e88782ef8e10f060041c6067cf706fa7447a2f28bb49833f9fc475606aR1012

Reply:
great catch, I added the patch, thanks!